homeassistant/components/opentherm_gw/sensor.py

```diff
@@ -1,20 +1,621 @@
 """Support for OpenTherm Gateway sensors."""
 
-import logging
+from dataclasses import dataclass
 
-from homeassistant.components.sensor import ENTITY_ID_FORMAT, SensorEntity
+import pyotgw.vars as gw_vars
+
+from homeassistant.components.sensor import (
+    ENTITY_ID_FORMAT,
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
 from homeassistant.config_entries import ConfigEntry
-from homeassistant.const import CONF_ID
+from homeassistant.const import (
+    CONF_ID,
+    PERCENTAGE,
+    UnitOfPower,
+    UnitOfPressure,
+    UnitOfTemperature,
+    UnitOfTime,
+    UnitOfVolumeFlowRate,
+)
 from homeassistant.core import HomeAssistant, callback
-from homeassistant.helpers.device_registry import DeviceInfo
-from homeassistant.helpers.dispatcher import async_dispatcher_connect
 from homeassistant.helpers.entity import async_generate_entity_id
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
-from . import DOMAIN
-from .const import DATA_GATEWAYS, DATA_OPENTHERM_GW, SENSOR_INFO, TRANSLATE_SOURCE
+from . import OpenThermGatewayDevice
+from .const import DATA_GATEWAYS, DATA_OPENTHERM_GW
+from .entity import OpenThermEntity, OpenThermEntityDescription
+
+SENSOR_FLOAT_SUGGESTED_DISPLAY_PRECISION = 1
+
+
+@dataclass(frozen=True, kw_only=True)
+class OpenThermSensorEntityDescription(
+    SensorEntityDescription, OpenThermEntityDescription
+):
+    """Describes opentherm_gw sensor entity."""
+
 
-_LOGGER = logging.getLogger(__name__)
+SENSOR_INFO: tuple[tuple[list[str], OpenThermSensorEntityDescription], ...] = (
+    (
+        [gw_vars.BOILER, gw_vars.THERMOSTAT],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.DATA_CONTROL_SETPOINT,
+            friendly_name_format="Control Setpoint {}",
+            device_class=SensorDeviceClass.TEMPERATURE,
+            state_class=SensorStateClass.MEASUREMENT,
+            native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+            suggested_display_precision=SENSOR_FLOAT_SUGGESTED_DISPLAY_PRECISION,
+        ),
+    ),
+    (
+        [gw_vars.BOILER, gw_vars.THERMOSTAT],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.DATA_MASTER_MEMBERID,
+            friendly_name_format="Thermostat Member ID {}",
+        ),
+    ),
+    (
+        [gw_vars.BOILER, gw_vars.THERMOSTAT],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.DATA_SLAVE_MEMBERID,
+            friendly_name_format="Boiler Member ID {}",
+        ),
+    ),
```

Comment:
What does this return?

Reply:
Those 2 sensors return the member IDs as reported by the thermostat and the boiler, respectively. From what I gather so far it seems to be an ID that the OpenTherm Association assigns to its members (equipment manufacturers).
In my setup, the thermostat reports member ID `13` (99% sure it's a Honeywell) and the boiler returns `24`(Vaillant I believe), I can't verify the manufacturers right now as I won't be home for another while.
It is a value between `0` and `255`, where ID `0` is reserved for `a customer non-specific device`. I have not been able to find a complete list of IDs and manufacturers as the OpenTherm Association is remarkably closed for an entity with a name like that...

Reply:
I think these sensors can be deprecated in the future to be replaced with a device info manufacturer. You could use logging in the library to identify new member IDs and complete the list

Reply:
What do you mean by `logging`? Like notifying users through HA and asking them to report their values and manufacturers?

Reply:
Yep, I do this in Withings, since they have some undocumented values, so if the value is not in the enum, I will log a message asking people to open an issue in the lib repo